### PR TITLE
Put things together

### PR DESCRIPTION
**Pull request by CsengerG**
With this PR:
- the Android app can send the survey data using the JSON API requests (not encrypted yet)
  - I added a button to the toolbar which calls things to send the survey data
- the front-end appears to be working
  - it's a bit slow
  - but I hope this unblocks many things on the front-end thing
- a filter API for the name is added

So... I guess.. We are very very close to the MVP?

Further things to do (not in this PR):
- Look into encryption
- <del>Probably refactor the Java mess I created
  - it would be pretty nice if could distinguish easily between Boolean (i.e. yes-no) question, one-choice and multiple choice questions -- I would suggest that we create separate classes for them (**this was done since**)</del>

### CODE REVIEW

File: Survey/app/src/main/java/com/example/feverfinder/SurveyActivity.java [hunk 1]
```diff
@@ -144,4 +173,39 @@ public boolean onNavigationItemSelected(MenuItem item) {
     public void onFragmentInteraction(Uri uri) {
 
     }
+
+    private void saveAnswers() throws SaveException, JSONException, IOException {
+        // TODO: test for internet connection;
+        // TODO: if there isn't one save to file, or throw exception?
```
Comment (pmernyei) [inline Survey/app/src/main/java/com/example/feverfinder/SurveyActivity.java:179]: These TODO comments seem outdated, right?

File: Survey/app/src/main/java/com/example/feverfinder/questions/TextQuestion.java [hunk 2]
```diff
@@ -6,21 +6,25 @@
...
         super(name, label);
-        content = "";
     }
 
+    @Override
+    public Object getJSONOutput() {
+        return content;
+    }
 
     //TODO: don't want these requires APIs
```
Comment (pmernyei) [inline Survey/app/src/main/java/com/example/feverfinder/questions/TextQuestion.java:29]: Do we know what we're doing about this? (I know it's nothing to do with this PR though but want to note this...)

File: front_end/src/searchinput.js [hunk 3]
```diff
@@ -66,11 +66,16 @@ class SearchInput extends React.Component {
...
         var request = new XMLHttpRequest();
-        request.open('GET', 'http://127.0.0.1:8000/api/people/get_by_id/?id=7d516f50-02bf-40c3-8e83-5bc4321b8861', true);
+        //request.open('GET', 'http://127.0.0.1:8000/api/people/get_by_id/?id=7d516f50-02bf-40c3-8e83-5bc4321b8861', true);
+        request.open('GET', query, true);
         request.onload = (callback => function() {
-            var ps = JSON.parse(this.response);
-            ps[0].name = "Some Name";
-            callback(ps);
+            if(this.response === undefined || this.response.length == 0) {
+                // TODO: nothing found
```
Comment (pmernyei) [inline front_end/src/searchinput.js:73]: I'll make a quick PR to sort that once this is in

**Review by pmernyei — APPROVED** [review]
Seems great as far as I can tell, maybe remove those TODOs that seem outdated to me in the app code, then let's get this merged? Esp since pretty much any more work depends on it...

File: Survey/app/src/main/java/com/example/feverfinder/questions/TextQuestion.java (hunk 2 above)
Comment (gezacsenger) [reply Survey/app/src/main/java/com/example/feverfinder/questions/TextQuestion.java:29]: I don't even know what that means. Does it bother us?

**Comment by gezacsenger** [comment]
I'm closing this PR. [See the equivalent one.](https://github.com/mdaudali/FeverFinder/pull/14)

**Pull request closed without merging**